### ISSUE TITLE
Try to preserve expanded sessions when de-compacting sessions

### ISSUE THREAD

**Issue by connor4312** [microsoft/vscode#101438]
Testing #101009

I made a change in js-debug to `noCompact` 2nd level sessions (e.g. web workers and child processes), which is great!

However, the experience when opening a new tab (a 1st level session) is kind of weird. Here's a video: https://memes.peet.io/img/20-06-cb2e1fc9-f73a-46d5-8f8e-4ae5ddd588f0.webm The top-level node collapses, and previous session is now a child of the top-level node, also collapsed.

This will always be a _little_ bit of a weird experience, but I think some algorithm like this would make the experience better:

 1. Record all collapsed/expanded debug sessions
 2. Mutate the tree
 3. Re-apply all collapsed/expanded sessions
 4. Expand the tree such that whatever session was previously on top is still visible.

(Or, in simpler terms, expand the tree such that all previously-visible sessions are still visible)

**Comment by isidorn** [comment]
Thanks for the video. I suggest it is a bit weird.
However the callStack tree passes the `autoExpandSingleChildren` flag to the tree [here](https://github.com/microsoft/vscode/blob/b16b467d3e03e1a1ae05b5836e4e5a5af504e86d/src/vs/workbench/contrib/debug/browser/callStackView.ts#L230)

@joaomoreno I wonder why this flag does not seem to get respected in the attached video? Is there some corner case for the flag, or single children should always get auto expanded?

**Comment by joaomoreno** [comment]
Post change, there is no single children sequence that can automatically be expanded. The first parent has two children.

That flag is also irrelevant to the collapsing. I suggest to place some breakpoints and see the full changes to the tree. Figure out why that top level node gets collapsed.

**Comment by isidorn** [comment]
@joaomoreno ok.
However @connor4312 can you try this again please after my fix for https://github.com/microsoft/vscode/issues/101431

**Comment by connor4312** [comment]
This is better now, things stay expanded!
